pkg/sdn/plugin/plugin.go

```diff
@@ -1,65 +0,0 @@
-package plugin
-
-import (
-	"fmt"
-
-	utilsets "k8s.io/apimachinery/pkg/util/sets"
-	"k8s.io/kubernetes/pkg/apis/componentconfig"
-	kubeletTypes "k8s.io/kubernetes/pkg/kubelet/container"
-	knetwork "k8s.io/kubernetes/pkg/kubelet/network"
-	kcni "k8s.io/kubernetes/pkg/kubelet/network/cni"
-
-	"github.com/golang/glog"
-)
-
-// This kubelet network plugin shim only exists to grab the knetwork.Host
-// Everything else is simply proxied directly to the kubenet CNI driver.
-func (node *OsdnNode) Init(host knetwork.Host, hairpinMode componentconfig.HairpinMode, nonMasqueradeCIDR string, mtu int) error {
-	plugins := kcni.ProbeNetworkPlugins(kcni.DefaultNetDir, kcni.DefaultCNIDir)
-	if len(plugins) == 0 {
-		return fmt.Errorf("openshift-sdn CNI network plugin config could not be found in %v", kcni.DefaultNetDir)
-	} else if len(plugins) > 1 {
-		return fmt.Errorf("multiple CNI network plugins found; only one can be used")
-	}
-	node.host = host
-	node.kubeletCniPlugin = plugins[0]
-
-	err := node.kubeletCniPlugin.Init(host, hairpinMode, nonMasqueradeCIDR, mtu)
-
-	// Let initial pod updates happen if they need to
-	glog.V(5).Infof("openshift-sdn CNI plugin initialized")
-	close(node.kubeletInitReady)
-
-	return err
-}
-
-func (node *OsdnNode) Name() string {
-	return kcni.CNIPluginName
-}
-
-func (node *OsdnNode) Capabilities() utilsets.Int {
-	return utilsets.NewInt(knetwork.NET_PLUGIN_CAPABILITY_SHAPING)
```

Comment:
The real CNI plugin does not set this...

pkg/kubelet/kubelet_network.go:shapingEnabled() helpfully observes

        // ... The combination of (cri + network plugin that
        // handles bandwidth shaping) may not work because of this.